src/purify.ts

```diff
@@ -2022,3 +2028,5 @@ export type WindowLike = Pick<
   document?: Document;
   MozNamedAttrMap?: typeof window.NamedNodeMap;
 } & Pick<TrustedTypesWindow, 'trustedTypes'>;
+
+type ArrayElement<T> = T extends Array<infer U> ? U : never;
```

Comment:
This is a utility type that extracts each element type of a union type composed of an array.

```ts
type test = ArrayElement<string[]>; // string
type test2 = ArrayElement<number[]>; // number
type test3 = ArrayElement<string[] | number[]>; // string | number
```